serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc

```diff
@@ -0,0 +1,359 @@
+     "transition": "FinalizeWithError"
+   }
+ ],
+ "timeouts": {
+   "eventTimeout": "PT30S"
+ }
+}
+----
+
+=== Switch state timeout
+
+Switch states can be defined with link:{spec_doc_url}#switch-state-event-conditions[eventConditions] property, where the workflow execution waits to make a decision depending on the events to be consumed and matched, defined through link:{spec_doc_url}#event-definition[event definition].
+
+In this situation, you can define an event timeout, that controls the maximum time to wait for an event to match the conditions, if this time is expired, the workflow moves to the state defined in the `defaultCondition` property in the switch state, as showed in the <<switch-state, example>>.
+
+See more details about this state on the link:{spec_doc_url}#switch-date[Serverless Workflow specification].
+
+[#switch-state]
+.Example of switch state with timeout
+[source,json]
+----
+{
+    "name": "ChooseOnEvent",
+    "type": "switch",
+    "eventConditions": [
+    {
+        "eventRef": "visaApprovedEvent",
+        "transition": "ApprovedVisa"
+    },
+    {
+        "eventRef": "visaDeniedEvent",
+        "transition": "DeniedVisa"
+    }
+    ],
+        "defaultCondition": {
+        "transition": "HandleNoVisaDecision"
+    },
+        "timeouts": {
+        "eventTimeout": "PT5S"
+    }
+}
+----
+
+=== Deploying a timed-based workflow
+
+n order to deploy a work that contains timeouts or any other timer-based action, it is necessary to have Job Service running in your environment, which is an external service responsible for workflows timers, see the <<job-service, section>> for more information.
+In the <<timeout-example, timeout example>> you can see in details how to set up a knative infrastructure with the workflow and job service running.
+
+[#job-service]
+=== Job Service configuration
+
+All timer-related actions that might be declared in a workflow, are handled by a supporting service, called Job Service, which is responsible for managing, scheduling, and firing all actions (jobs) to be executed in the workflows.
+
+Suppose the workflow service is not configured to use job service or there is no such service running. In that case, all timer-related actions use an embedded in-memory implementation of job service, which should not be used in production, since when the application shutdown, all timers are lost, which in a serverless architecture is a very common behavior with the scale to zero approach. That said, the no job service configuration can only be used for testing or development, but not for production.
+
+The main goal of the Job Service is to work with only active jobs. The Job Service tracks only the jobs that are scheduled and that need to be executed. When a job reaches a final state, the job is removed from the Job Service. All job information and transition states are sent to the Kogito Data Index Service where they can be indexed and made available for GraphQL queries.
```

Comment:
When configured, all the jobs information and status changes are sent to the Kogito Data Index Service, where they can be indexed and made available for GraphQL queries.

Reply:
thanks